packages/react-scripts/config/webpack.config.prod.js

```diff
@@ -413,43 +413,21 @@ module.exports = {
     // having to parse `index.html`.
     new ManifestPlugin({
       fileName: 'asset-manifest.json',
-      publicPath: publicPath
-    }),
-    // Generate a service worker script that will precache, and keep up to date,
-    // the HTML & assets that are part of the Webpack build.
-    new SWPrecacheWebpackPlugin({
-      // By default, a cache-busting query parameter is appended to requests
-      // used to populate the caches, to ensure the responses are fresh.
-      // If a URL is already hashed by Webpack, then there is no concern
-      // about it being stale, and the cache-busting can be skipped.
-      dontCacheBustUrlsMatching: /\.\w{8}\./,
-      filename: 'service-worker.js',
-      logger(message) {
-        if (message.indexOf('Total precache size is') === 0) {
-          // This message occurs for every build and is a bit too noisy.
-          return;
-        }
-        if (message.indexOf('Skipping static resource') === 0) {
-          // This message obscures real errors so we ignore it.
-          // https://github.com/facebook/create-react-app/issues/2612
-          return;
-        }
-        console.log(message);
-      },
-      minify: true,
-      // Don't precache sourcemaps (they're large) and build asset manifest:
-      staticFileGlobsIgnorePatterns: [/\.map$/, /asset-manifest\.json$/],
-      // `navigateFallback` and `navigateFallbackWhitelist` are disabled by default; see
-      // https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#service-worker-considerations
-      // navigateFallback: publicUrl + '/index.html',
-      // navigateFallbackWhitelist: [/^(?!\/__).*/],
+      publicPath: publicPath,
     }),
     // Moment.js is an extremely popular library that bundles large locale files
     // by default due to how Webpack interprets its code. This is a practical
     // solution that requires the user to opt into importing specific locales.
     // https://github.com/jmblog/how-to-optimize-momentjs-with-webpack
     // You can remove this if you don't use Moment.js:
     new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/),
+    // Generate a service worker script that will precache, and keep up to date,
+    // the HTML & assets that are part of the Webpack build.
+    new WorkboxWebpackPlugin.GenerateSW({
+      exclude: [/\.map$/, /^(?:asset-)manifest.*\.js(?:on)?$/],
+      navigateFallback: '/index.html',
+      navigateFallbackWhitelist: [/^(?!\/__).*/], // fix for Firebase
```

Comment:
`navigateFallbackBlacklist` is newly available in `workbox-webpack-plugin`, to address the awkwardness of needing to put in a negative pattern here.

We could also potentially get a little more nuanced, and work around some of the issues raised in https://github.com/facebook/create-react-app/issues/3008#issuecomment-325156465 (which led to `navigateFallback` being turned off by default) with the blacklist. One approach would be to blacklist any URLs whose last part of the path contains a `.` character, since that's likely to be a file extension, and those have historically been files that should not be included in the `navigateFallback` logic.

And example config could be:

```js
{
  // ...other options...
  navigateFallback: '/index.html',
  navigateFallbackBlacklist: [
    new RegExp('^/_'),
    new RegExp('/[^/]+\.[^/]+$'),
  ],
}
```

See https://regex101.com/r/7f207Y/1

Reply:
Got it. Should the `.` char RegExp maybe be `/[^/]+\.[^/]+/?$` to include a potential trailing slash?

Reply:
I think if there's a trailing slash, we *don't* want that to be blacklisted. I.e. we *do* want the `navigateFallback` behavior to take effect.

Reply:
Ok

Reply:
Also, in the current implementation, the firebase url is a double __ rather than single. Is that correct? If so should the single underscore prefixed urls also be blacklisted?

Reply:
I think it's safe to go with blacklisting anything that starts with `/_`, i.e. a single underscore. I don't know how much of a convention it is to use a single vs. a double for "special" URLs outside of Firebase, but the false-positive rate of checking for a single underscores seems likely to be acceptable.